woocommerce/google-listings-and-ads
# Integrate Storybook

### Changes proposed in this Pull Request:

This PR adds Story Book to the project.

### Screenshots:

<img width="922" alt="Screenshot 2022-03-29 at 10 25 03" src="https://user-images.githubusercontent.com/5908855/160567580-67e825f9-8f5b-4559-bc30-34ed8532e456.png">

### Detailed test instructions:

1. Checkout this branch
2. Run `npm i` to install all the new dependencies
3. Run `npm run storybook` in order to create a local instance of storybook
4. Inside the StoryBookpage, check the Tree Select Component
5. Run `npm run storybook:build`
6. See how it creates a static storybook instance inside `.storybook/dist`
7. Go to `https://woocommerce.github.io/google-listings-and-ads/?path=/story/tree-select-control--base` 
8. You can see the StoryBook public page
9. Edit `js/src/components/tree-select-control/stories/index.js` (for example in line 79, change ES to FR
10. Run `npm run storybook:deploy` to publish a new version of the public StoryBook
11. Go to `https://woocommerce.github.io/google-listings-and-ads/?path=/story/tree-select-control--base`  and see the updated changes 


### Additional details:

- `/.storybook/main.js` has the main config for the storybook
- `/.storybook/preview.js` loads the general styles
- `/.storybook/styles.scss` WP/WC specific styles

You can create stories inside `js/src` folder, just adding the folder stories inside a component.

- Removed `color-studio` dependency and component test page since they are not anymore necessary.
- Tweak some style variables
